src/cache/cache_las.c

```diff
@@ -1423,10 +1433,50 @@ __wt_find_lookaside_upd(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDAT
         upd->durable_ts = durable_timestamp;
         upd->start_ts = las_timestamp;
         upd->prepare_state = prepare_state;
-        /*
-         * Mark this update as external and to be discarded when not needed.
-         */
-        upd->ext = 1;
+
+        if (prepare_state == WT_PREPARE_INPROGRESS) {
+            WT_ASSERT(session, !modify);
+            WT_ASSERT(session, cbt->compare == 0);
+            upd_entry = NULL;
+            switch (page->type) {
+            case WT_PAGE_COL_FIX:
+            case WT_PAGE_COL_VAR:
+                WT_ASSERT(session, cbt->ins != NULL);
+                upd_entry = &cbt->ins->upd;
+                break;
+            case WT_PAGE_ROW_LEAF:
+                upd_entry =
```

Comment:
@kommiharibabu My understanding is that col/row instantiation has already happened so I can rely on these here. Does this look right to you?

Reply:
Yes, the page has instantiated with all the keys by this time.